src/Simulations/simulation.jl

```diff
@@ -91,15 +91,7 @@ function set!(simulation::Simulation, subcomponent = :base; kwargs...)
 
         # sometimes we want to set with a function and sometimes with an
         # Field. This supports both behaviors.
-        if f isa Function
-            space = axes(target_field)
-            local_geometry = Fields.local_geometry_field(space)
-            target_field .= f.(local_geometry)
-        elseif f isa Fields.Field # ClimaCore Field
-            target_field .= f
```

Comment:
Here it gives the flexibility to initialize from either a know function (of locations) or a known fields